File: Documentation/3.5/Samples/csharp/Raven.Documentation.Samples/ClientApi/HowTo/SendCustomRequest.cs
using System.Net.Http;
using Raven.Abstractions.Data;
using Raven.Client.Connection;
using Raven.Client.Connection.Implementation;
using Raven.Client.Document;
using Raven.Json.Linq;

namespace Raven.Documentation.Samples.ClientApi.HowTo
{
	public class SendCustomRequest
	{
		public SendCustomRequest()
		{
			using (var store = new DocumentStore())
			{
				#region custom_request_1
				string key = "employees/1";

				// http://localhost:8080/databases/Northwind/docs/employees/1
				string url = store.Url // http://localhost:8080
					.ForDatabase("Northwind") // /databases/Northwind
					.Doc(key); // /docs/employees/1

				IDatabaseCommands commands = store.DatabaseCommands;
				using (HttpJsonRequest request = store
					.JsonRequestFactory
					.CreateHttpJsonRequest(new CreateHttpJsonRequestParams(commands, url, HttpMethod.Get, commands.PrimaryCredentials, store.Conventions)))
				{
					RavenJToken json = request.ReadResponseJson();
					JsonDocument jsonDocument = SerializationHelper
						.DeserializeJsonDocument(key, json, request.ResponseHeaders, request.ResponseStatusCode);
				}
				#endregion
			}
		}
	}
}

File: Documentation/3.5/Samples/java/src/test/java/net/ravendb/clientapi/howto/SendCustomRequest.java
package net.ravendb.clientapi.howto;

import net.ravendb.abstractions.data.HttpMethods;
import net.ravendb.abstractions.data.JsonDocument;
import net.ravendb.abstractions.json.linq.RavenJToken;
import net.ravendb.client.connection.CreateHttpJsonRequestParams;
import net.ravendb.client.connection.IDatabaseCommands;
import net.ravendb.client.connection.RavenUrlExtensions;
import net.ravendb.client.connection.SerializationHelper;
import net.ravendb.client.connection.implementation.HttpJsonRequest;
import net.ravendb.client.document.DocumentStore;


public class SendCustomRequest {
  @SuppressWarnings("unused")
  public SendCustomRequest() throws Exception {
    try (DocumentStore store = new DocumentStore()) {
      //region custom_request_1
      String key = "employees/1";

      // http://localhost:8080/databases/Northwind/docs/employees/1
      String forDatabase = RavenUrlExtensions.forDatabase(store.getUrl(), "Northwind");
      String url = forDatabase + "/docs/" + key;

      IDatabaseCommands commands = store.getDatabaseCommands();
      HttpJsonRequest request = store
        .getJsonRequestFactory()
        .createHttpJsonRequest(new CreateHttpJsonRequestParams(commands, url, HttpMethods.GET, null, commands.getPrimaryCredentials(), store.getConventions()));

      RavenJToken json = request.readResponseJson();
      JsonDocument jsonDocument = SerializationHelper.deserializeJsonDocument(key, json, request.getResponseHeaders(), request.getResponseStatusCode());
      //endregion
    }
  }
}
